pkg/types/builder.go

```diff
@@ -155,19 +155,25 @@ func (g *Builder) buildResource(res *schema.Resource, cfg *config.Resource, tfPa
 			sfx := "SecretRef"
 			cfg.Sensitive.AddFieldPath(fieldPathWithWildcard(tfPaths), "spec.forProvider."+fieldPathWithWildcard(xpPaths)+sfx)
 			// todo(turkenh): do we need to support other field types as sensitive?
-			if fieldType.String() != "string" && fieldType.String() != "*string" {
-				return nil, nil, fmt.Errorf(`got type %q for field %q, only types "string" and "*string" supported as sensitive`, fieldType.String(), fieldNameCamel)
+			if fieldType.String() != "string" && fieldType.String() != "*string" && fieldType.String() != "[]string" &&
+				fieldType.String() != "[]*string" {
+				return nil, nil, fmt.Errorf(`got type %q for field %q, only types "string", "*string", []string and []*string supported as sensitive`, fieldType.String(), fieldNameCamel)
```

Comment:
I remember @muvaf was [asking for](https://github.com/crossplane/terrajet/issues/100#issuecomment-974136406) `map[string]*string` support as well. 
Did you consider including that into this PR? Otherwise, I would suggest to create an issue for that while closing the original one with this PR.

Reply:
@turkenh  Thank you for this comment. As we discusses offline, providing the `map[string]*string` in a separate PR will be easier in terms of reviewing, testing and merging them.

I will open a separate PR and it will fixes the following issue (this issue is about that map[string]*string support): https://github.com/crossplane/terrajet/issues/208